components/form/Form.tsx

```diff
@@ -52,7 +52,7 @@ export type ValidationRule = {
   /** transform a value before validation */
   transform?: (value: any) => any;
   /** custom validate function (Note: callback must be called) */
-  validator?: (rule: any, value: any, callback: any) => any;
+  validator?: (rule: any, value: any, callback: any, source?: any, options?: any) => any;
```

Comment:
@paranoidjk you can PR to do so, until we can provide more specific definition for source and options.